featuretools/feature_base/feature_base.py

```diff
@@ -333,94 +328,95 @@ def __invert__(self):
         return self.NOT()
 
     def unique_name(self):
-        return u"%s: %s" % (self.entity_id, self.get_name())
+        return u"%s: %s" % (self.dataframe_name, self.get_name())
 
     def relationship_path_name(self):
         return self.relationship_path.name
 
 
 class IdentityFeature(FeatureBase):
-    """Feature for entity that is equivalent to underlying variable"""
-
-    def __init__(self, variable, name=None):
-        entity_id = variable.entity_id
-        self.variable = variable.entityset.metadata[entity_id][variable.id]
-        self.return_type = type(variable)
-        super(IdentityFeature, self).__init__(entity=variable.entity,
+    """Feature for dataframe that is equivalent to underlying column"""
+
+    def __init__(self, entityset, dataframe_name, column_name, name=None):
+        self.column_name = column_name
+        column_accessor = entityset[dataframe_name].ww[column_name]
+        self.return_type = column_accessor.ww.schema
+        super(IdentityFeature, self).__init__(entityset=entityset,
+                                              dataframe_name=dataframe_name,
                                               base_features=[],
                                               relationship_path=RelationshipPath([]),
                                               primitive=PrimitiveBase,
                                               name=name)
 
     @classmethod
     def from_dictionary(cls, arguments, entityset, dependencies, primitives_deserializer):
```

Comment:
Do we need `dependencies` and `primitives_deserializer` as parameters here?

Reply:
I believe it's so the `from_dictionary` arguments are uniform across feature types so the feature deserializer doesn't have to think about what arguments to pass